### PR TITLE
Refactored LiveJobs update startAt logic

### PR DESCRIPTION
**Pull request by marcocast**

### CODE REVIEW

**Comment by tobwiens** [comment]
:+1: Nice, I like it!

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/StartAtUpdater.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            genericInformation.put(ExtendedSchedulerPolicy.GENERIC_INFORMATION_KEY_START_AT, startAt);
+            job.setGenericInformations(genericInformation);
+            return true;
+        }
+        return false;
+    }
+
+    private boolean iaAValidStartAt(Map<String, String> genericInformation, String startAt) {
+        if (genericInformation.containsKey(ExtendedSchedulerPolicy.GENERIC_INFORMATION_KEY_START_AT)) {
+            if (genericInformation.get(ExtendedSchedulerPolicy.GENERIC_INFORMATION_KEY_START_AT)
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/StartAtUpdater.java]: MAJOR Merge this if statement with the enclosing one. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1066)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/StartAtUpdater.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+public class StartAtUpdater {
+
+    public boolean updateStartAt(InternalJob job, String startAt, SchedulerDBManager dbManager) {
+
+        long scheduledTime = ISO8601DateUtil.toDate(startAt).getTime();
+
+        Set<TaskId> updatedTasks = updateStartAtAndTasksScheduledTime(job, startAt, scheduledTime);
+
+        if (updatedTasks.size() > 0) {
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/helpers/StartAtUpdater.java]: MAJOR Use isEmpty() to check whether the collection is empty or not. [rule](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1155)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Pull request merged**